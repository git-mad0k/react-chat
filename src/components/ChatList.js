import React from 'react'

import List from '@material-ui/core/List'
import { withStyles } from '@material-ui/core/styles'

import ChatItem from './ChatItem'

const styles = theme => ({
  chatList: {
    height: `calc(100% - 56px)`,
    overflowY: 'scroll',
    flex: '1 1 auto'
  },
})

const ChatList = ({ classes, chats }) => (
  <List className={classes.chatList}>
      {chats.map((chat, i) => (
        <ChatItem chat={chat} key={i} />        
      ))}
  </List>
)

export default withStyles(styles)(ChatList)
