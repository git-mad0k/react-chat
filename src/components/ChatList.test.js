/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ChatList from './ChatList';

jest.mock('./ChatItem.js', () => ({ chatId, title }) => `ChatId-${chatId} ChatTitlte-${title}`);

const mockProps = {
  chats: [
    {
      _id: '1',
      title: 'test1',
      createdAt: '2018-03-16T10:53:23.200Z',
    },
    {
      _id: '2',
      title: 'test2',
      createdAt: '2018-03-16T10:53:23.200Z',
    },
    {
      _id: '3',
      title: 'test3',
      createdAt: '2018-03-16T10:53:23.200Z',
    },
  ],
  activeChat: {
    createdAt: '2018-03-16T10:53:23.200Z',
    creator: {
      _id: '111',
      username: 'test',
      firstName: 'test',
      lastName: 'test',
    },
    title: 'test1',
  },
  disabled: false,
};
describe('<ChatList />', () => {
  it('render without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ChatList {...mockProps} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders correctly', () => {
    const tree = renderer.create(<ChatList {...mockProps} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders without chats', () => {
    const tree = renderer.create(<ChatList {...mockProps} chats={[]} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
