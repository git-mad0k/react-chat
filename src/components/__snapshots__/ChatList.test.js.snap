// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatList /> renders correctly 1`] = `
<ul
  className="MuiList-root-9 MuiList-padding-10 ChatList-chatList-7"
>
  ChatId-1 ChatTitlte-test1
  ChatId-2 ChatTitlte-test2
  ChatId-3 ChatTitlte-test3
</ul>
`;

exports[`<ChatList /> renders without chats 1`] = `
<ul
  className="MuiList-root-9 MuiList-padding-10 ChatList-chatList-7"
>
  <h3
    className="MuiTypography-root-13 MuiTypography-subheading-20 ChatList-noChats-8"
  >
    There is no chats yet...
  </h3>
</ul>
`;
